src/parser/json/package_parser.py

```diff
@@ -0,0 +1,181 @@
+from src.parser.json.actor_parser import ActorParser
+from src.parser.json.checksum_parser import ChecksumParser
+from src.parser.json.dict_parsing_functions import append_parsed_field_or_log_error, datetime_from_str, \
+    raise_parsing_error_if_logger_has_messages, json_str_to_enum_name, construct_or_raise_parsing_error, \
+    parse_field_or_log_error, parse_field_or_no_assertion_or_none, parse_field_or_no_assertion, parse_list_of_elements
+from src.parser.json.license_expression_parser import LicenseExpressionParser
+from src.parser.logger import Logger
+
+
+class PackageParser:
+    logger: Logger
+    actor_parser: ActorParser
+    checksum_parser: ChecksumParser
+    license_expression_parser: LicenseExpressionParser
+
+    def __init__(self):
+        self.actor_parser = ActorParser()
+        self.checksum_parser = ChecksumParser()
+        self.license_expression_parser = LicenseExpressionParser()
+        self.logger = Logger()
+
+    def parse_package(self, package_dict: Dict) -> Package:
+        logger = Logger()
+        name: Optional[str] = package_dict.get("name")
+        spdx_id: Optional[str] = package_dict.get("SPDXID")
+        attribution_texts: List[str] = package_dict.get("attributionTexts", [])
+
+        built_date: Optional[datetime] = parse_field_or_log_error(logger, package_dict.get("builtDate"),
+                                                                  datetime_from_str)
+
+        checksums = parse_field_or_log_error(logger, package_dict.get("checksums"),
+                                             lambda x: parse_list_of_elements(self.checksum_parser.parse_checksum, x))
+        comment: Optional[str] = package_dict.get("comment")
+        copyright_text: Optional[str] = package_dict.get("copyrightText")
+        description: Optional[str] = package_dict.get("description")
+        download_location: Optional[Union[str, SpdxNoAssertion, SpdxNone]] = parse_field_or_no_assertion_or_none(
+            package_dict.get("downloadLocation"))
+
+        external_refs: List[ExternalPackageRef] = parse_field_or_log_error(logger, package_dict.get("externalRefs"),
+                                                                           self.parse_external_refs)
+
+        files_analyzed: Optional[bool] = parse_field_or_log_error(logger, package_dict.get("filesAnalyzed"),
+                                                                  lambda x: x, True)
+        homepage: Optional[str] = package_dict.get("homepage")
+        license_comments: Optional[str] = package_dict.get("licenseComments")
+        license_concluded = parse_field_or_log_error(logger, package_dict.get("licenseConcluded"),
+                                                     lambda x: parse_field_or_no_assertion_or_none(x,
+                                                                                                   self.license_expression_parser.parse_license_expression),
+                                                     None)
```

Comment:
This looks weird. Methods down below as well

Reply:
fixed